Compile.ps1

```diff
@@ -3,29 +3,39 @@ param (
     [switch]$Run,
     [switch]$SkipPreprocessing
 )
+
 $OFS = "`r`n"
 $scriptname = "winutil.ps1"
 $workingdir = $PSScriptRoot
+$logFilePath = Join-Path $workingdir "compile.log"
 
 # Variable to sync between runspaces
 $sync = [Hashtable]::Synchronized(@{})
 $sync.PSScriptRoot = $workingdir
 $sync.configs = @{}
 
-function Update-Progress {
-    param (
-        [Parameter(Mandatory, position=0)]
-        [string]$StatusMessage,
+# Dot-source external functions
+. "$PSScriptRoot\tools\Update-Progress.ps1"
```

Comment:
@fam007e you've stated in your PR description and in this line that there's indeed a script file with the name of `Update-Progress.ps1`, but in your **PR changes** there's no such file (you're only editing `Compile.ps1` Script).

Make sure to add this newly created file with Git CLI (or using GitHub Desktop if you're using it). You can add `tools/Update-Progress.ps1` file with the following git commands (if you've Git CLI installed):
```ps1
git add ./tools/Update-Progress.ps1
git commit -m "Add 'Update-Progress' Script Utility"
```